handler.js

```diff
@@ -277,13 +277,57 @@ export function ingest({ Records: records }: Object, context: Object) {
           return parse(raw)
             .then(mapToAirings)
             .then(function(airings) {
-
               console.log('Parsed airings list', airings.length);
 
+              let airingData = {};
+
+              airings.forEach(function (a: Airing) {
+                if (airingData[a.channel] === undefined) {
+                  airingData[a.channel] = {
+                    channel: a.channel,
+                    startTime: 0,
+                    endTime: 0,
+                    airings: []
+                  };
+                }
+                airingData[a.channel].airings.push(a.id);
+                if (airingData[a.channel].endTime === 0 || a.date < airingData[a.channel].startTime) {
+                  airingData[a.channel].startTime = a.date;
+                }
+                if (airingData[a.channel].endTime === 0 || a.end_date > airingData[a.channel].endTime) {
+                  airingData[a.channel].endTime = a.date;
+                }
+              });
+
+              let removeAiringPromises = [];
+
+              for (let a in airingData) {
+                if (airingData.hasOwnProperty(a)) {
+                  a = airingData[a];
+                  let searchPromise = actions.searchAirings({
```

Comment:
Given that the XML files that we ingest represent a sliding window (1 month back, 2 forward), along with us deleting airing info over time. Using a start / end time should be equivalent to selecting all airings over a channel.

You can use something like:
```
actions.searchAirings({
  term: p(event, 'term'),
  start: q(event, 'start') || 0,
  end: q(event, 'end') || Number.MAX_SAFE_INTEGER
}),
```

This way you don't need to try and compute a start and an end.